### PR TITLE
XML entities declared in a DTD are marked undeclared after XML file change

### PR DESCRIPTION
**Pull request by angelozerr**
XML entities declared in a DTD are marked undeclared after XML file change

Fixes https://github.com/redhat-developer/vscode-xml/issues/234

Signed-off-by: azerr <azerr@redhat.com>

### CODE REVIEW

**Comment by angelozerr** [comment]
Here a demo to test this PR:

EntityDemo

File: org.eclipse.lemminx/src/main/java/org/eclipse/lemminx/extensions/contentmodel/participants/diagnostics/LSPSAXParser.java [hunk 1]
```diff
@@ -0,0 +1,198 @@
...
+	private void init(LSPErrorReporterForXML reporter) {
+		try {
+			// Add LSP error reporter to fill LSP diagnostics from Xerces errors
+			super.setProperty("http://apache.org/xml/properties/internal/error-reporter", reporter);
+			super.setFeature("http://apache.org/xml/features/continue-after-fatal-error", false); //$NON-NLS-1$
+			super.setFeature("http://xml.org/sax/features/namespace-prefixes", true); //$NON-NLS-1$
+			super.setFeature("http://xml.org/sax/features/namespaces", true); //$NON-NLS-1$
+			super.setFeature("http://apache.org/xml/features/nonvalidating/load-external-dtd", true);
+		} catch (SAXNotRecognizedException | SAXNotSupportedException e) {
+			// Should never occurs.
```
Comment (xorye) [inline org.eclipse.lemminx/src/main/java/org/eclipse/lemminx/extensions/contentmodel/participants/diagnostics/LSPSAXParser.java:89]: // Should never occur

File: org.eclipse.lemminx/src/main/java/org/eclipse/lemminx/extensions/contentmodel/participants/diagnostics/LSPSAXParser.java [hunk 2]
```diff
@@ -0,0 +1,198 @@
...
+		if (systemId != null) {
+			// There a declared DTD in the DOCTYPE
+			// <!DOCTYPE root-element SYSTEM "./extended.dtd" []>
+			String eid = null;
+			try {
+				eid = XMLEntityManager.expandSystemId(systemId, locator.getExpandedSystemId(), false);
+			} catch (java.io.IOException e) {
+			}
+			if (!isDTDExists(eid)) {
+				// The declared DTD doesn't exists
```
Comment (xorye) [inline org.eclipse.lemminx/src/main/java/org/eclipse/lemminx/extensions/contentmodel/participants/diagnostics/LSPSAXParser.java:114]: // The declared DTD doesn't exist

File: org.eclipse.lemminx/src/main/java/org/eclipse/lemminx/extensions/contentmodel/participants/diagnostics/LSPSAXParser.java (hunk 1 above)
Comment (angelozerr) [reply org.eclipse.lemminx/src/main/java/org/eclipse/lemminx/extensions/contentmodel/participants/diagnostics/LSPSAXParser.java]: fixed

File: org.eclipse.lemminx/src/main/java/org/eclipse/lemminx/extensions/contentmodel/participants/diagnostics/LSPSAXParser.java (hunk 2 above)
Comment (angelozerr) [reply org.eclipse.lemminx/src/main/java/org/eclipse/lemminx/extensions/contentmodel/participants/diagnostics/LSPSAXParser.java]: fixed

**Comment by fbricon** [comment]
CI Build [fails](https://ci.eclipse.org/lemminx/blue/organizations/jenkins/lemminx/detail/PR-636/4/pipeline):

```
[ERROR] Failures: 

[ERROR]   DTDDiagnosticsTest.MSG_ELEMENT_NOT_DECLARED:39->testDiagnosticsFor:385 Unexpected diagnostics:

[] ==> iterable lengths differ, expected: <2> but was: <0>

[ERROR]   XMLSyntaxDiagnosticsTest.testAttributeNSNotUnique2:72 Unexpected diagnostics:

[Diagnostic [

  range = Range [

    start = Position [

      line = 2

      character = 1

    ]

    end = Position [

      line = 2

      character = 10

    ]

  ]

  severity = Error

  code = Either [

    left = AttributeNSNotUnique

    right = null

  ]

  source = "xsd"

  message = "Attribute "tns" bound to namespace "http://www.w3.org/2000/xmlns/" was already specified for element "xs:schema"."

  tags = null

  relatedInformation = null

], Diagnostic [

  range = Range [

    start = Position [

      line = 1

      character = 1

    ]

    end = Position [

      line = 1

      character = 10

    ]

  ]

  severity = Error

  code = Either [

    left = MarkupEntityMismatch

    right = null

  ]

  source = "xsd"

  message = "XML document structures must start and end within the same entity."

  tags = null

  relatedInformation = null

]] ==> iterable lengths differ, expected: <1> but was: <2>
```

**Comment by angelozerr** [comment]
> CI Build fails:

Fixed!

**Pull request merged**